recipes/mujoco/meta.yaml

```diff
@@ -0,0 +1,127 @@
+{% set name = "mujoco" %}
+{% set namecxx = "mujoco-cxx" %}
+{% set namepython = "mujoco-python" %}
+{% set version = "2.2.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/deepmind/mujoco/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 98cab2735817456c028e96ba34b2ee2c94cc4fab33ebcae15ab46fcc45a762ad
+  patches:
+    - python_remove_avx.patch
+    - python_remove_werror.patch
+    - disable_gcc_pragma.patch
+    - cxx_devendor.patch
+    - python_devendor.patch
+    - ccd_vec3_origin_link_problem.patch
+    - disable_copy_cxx_files_in_python_package.patch
+    - do_not_install_via_pip.patch
+
+
+build:
+  number: 0
+  skip: true  # [osx]
+
+outputs:
+  - name: {{ namecxx }}
+    script: build_cxx.sh  # [unix]
+    script: bld_cxx.bat  # [win]
+    build:
+      run_exports:
+        - {{ pin_subpackage(namecxx, max_pin='x.x.x') }}
+      # abseil-cpp is used only for tests, we can 
+      # safely ignore its run_exports
+      ignore_run_exports:
+        - abseil-cpp
```

Comment:
On Python (differently from C++) the library is used in the bindings themselfs, not on tests, see for example the Linux logs:
~~~
2022-06-18T14:40:23.0083572Z    INFO (mujoco-python,lib/python3.8/site-packages/mujoco/_structs.cpython-38-x86_64-linux-gnu.so): Needed DSO lib/libabsl_hash.so.2103.0.1 found in conda-forge::abseil-cpp-20210324.2-h9c3ff4c_0
2022-06-18T14:40:23.0147391Z    INFO (mujoco-python,lib/python3.8/site-packages/mujoco/_structs.cpython-38-x86_64-linux-gnu.so): Needed DSO lib/libabsl_raw_hash_set.so.2103.0.1 found in conda-forge::abseil-cpp-20210324.2-h9c3ff4c_0
2022-06-18T14:40:23.0176888Z    INFO (mujoco-python,lib/python3.8/site-packages/mujoco/_structs.cpython-38-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libpthread.so.0 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_15
2022-06-18T14:40:23.0238563Z    INFO (mujoco-python,lib/python3.8/site-packages/mujoco/_structs.cpython-38-x86_64-linux-gnu.so): Needed DSO lib/libmujoco.so.2.2.0 found in home/conda/staged-recipes/build_artifacts::mujoco-cxx-2.2.0-h696428e_0
2022-06-18T14:40:23.0298191Z    INFO (mujoco-python,lib/python3.8/site-packages/mujoco/_structs.cpython-38-x86_64-linux-gnu.so): Needed DSO lib/libabsl_throw_delegate.so.2103.0.1 found in conda-forge::abseil-cpp-20210324.2-h9c3ff4c_0
~~~

This is the reason why I did not check for that warning on Windows. However, the Windows warning is indeed fishy. By quickly looking at the abseil Windows package, in particular the size of the shipped `.lib`, at a first glance it seems that a library .lib is shipped, instead of a import library, however I need to check better.

Reply:
Ok, I checked a bit the situation. abseil-cpp is still pinned to a version that shipped static libraries on Windows (i.e. before https://github.com/conda-forge/abseil-cpp-feedstock/pull/25), see https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/a7759497a85d3c3a0e929583e5babacc922950c8/recipe/conda_build_config.yaml#L347 . Once an abseil with shared libraries is available, it will be used, and no modifications are required on the mujoco recipe. At this point, I think leaving avoiding to add the `ignore_run_exports` on the python side is probably the best/simplest thing to do.

Reply:
Agreed. If a future version provides non-static linking it's fine.